go.mod

```diff
@@ -4,18 +4,14 @@ go 1.12
 
 require (
 	github.com/gobuffalo/flect v0.1.5
-	github.com/golang/protobuf v1.3.1 // indirect
 	github.com/inconshreveable/mousetrap v1.0.0 // indirect
-	github.com/kr/pretty v0.1.0 // indirect
 	github.com/onsi/ginkgo v1.8.0
 	github.com/onsi/gomega v1.5.0
 	github.com/spf13/afero v1.2.2
 	github.com/spf13/cobra v0.0.3
 	github.com/spf13/pflag v1.0.3
-	golang.org/x/net v0.0.0-20190620200207-3b0461eec859 // indirect
 	golang.org/x/sys v0.0.0-20190621203818-d432491b9138 // indirect
-	golang.org/x/text v0.3.2 // indirect
 	golang.org/x/tools v0.0.0-20190614205625-5aca471b1d59
-	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127 // indirect
 	gopkg.in/yaml.v2 v2.2.2
+	k8s.io/apimachinery v0.0.0-20190927035529-0104e33c351d
```

Comment:
We were trying to avoid any dependencies on k8s, since what kubebuilder does is just scaffolding.
Since you only depend on one simple validation function in apimachinery, copy it over may not be a bad idea.

@droot WDYT?

Reply:
HI @mengqiy, 

IMHO the problem with is that who is responsible to do the validation is the k8s. So, if its change their and we are using it we will know before doing a new release. In POV is not a good idea decentralised the impl but let me know your final decision. 

c/c @droot @DirectXMan12

Reply:
The definition of DNS subdomains are unlikely to change unless it's a bugfix, and we'd basically just be pulling this in for  a regular expression.  That being said, the regular expression is non-trivial.

I think overall though it's probably worth just copying to avoid all the extra dependencies.

Reply:
(add a comment to just replace with library if we ever pull more stuff in)